internal/services/azurestackhci/stack_hci_marketplace_gallery_image_resource.go

```diff
@@ -0,0 +1,346 @@
+					},
+
+					"offer": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+
+					"sku": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+				},
+			},
+		},
+
+		"os_type": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(marketplacegalleryimages.OperatingSystemTypesLinux),
+				string(marketplacegalleryimages.OperatingSystemTypesWindows),
+			}, false),
+		},
+
+		"version": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"storage_path_id": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ForceNew:     true,
+			ValidateFunc: storagecontainers.ValidateStorageContainerID,
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (StackHCIMarketplaceGalleryImageResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r StackHCIMarketplaceGalleryImageResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 3 * time.Hour,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.AzureStackHCI.MarketplaceGalleryImages
+
+			var config StackHCIMarketplaceGalleryImageResourceModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			id := marketplacegalleryimages.NewMarketplaceGalleryImageID(subscriptionId, config.ResourceGroupName, config.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			payload := marketplacegalleryimages.MarketplaceGalleryImages{
+				Name:     pointer.To(config.Name),
+				Location: location.Normalize(config.Location),
+				Tags:     tags.Expand(config.Tags),
+				ExtendedLocation: &marketplacegalleryimages.ExtendedLocation{
+					Name: pointer.To(config.CustomLocationId),
+					Type: pointer.To(marketplacegalleryimages.ExtendedLocationTypesCustomLocation),
+				},
+				Properties: &marketplacegalleryimages.MarketplaceGalleryImageProperties{
+					Identifier: expandStackHCIMarketplaceGalleryImageIdentifier(config.Identifier),
+					OsType:     marketplacegalleryimages.OperatingSystemTypes(config.OsType),
+				},
+			}
+
+			if config.StoragePathId != "" {
+				payload.Properties.ContainerId = pointer.To(config.StoragePathId)
+			}
+
+			if config.HypervGeneration != "" {
```

Comment:
since `HypervGeneration` is required, it won't be empty?

Reply:
fixed